src/prefect/tasks.py

```diff
@@ -1033,14 +1037,15 @@ def map(
     ) -> PrefectFutureList[State[T]]:
         ...
 
+    @deprecated_async_method
     def map(
         self,
         *args: Any,
         return_state: bool = False,
         wait_for: Optional[Iterable[PrefectFuture]] = None,
         deferred: bool = False,
         **kwargs: Any,
-    ):
+    ) -> Iterable[Union[PrefectFuture, State]]:
```

Comment:
I think we need this typing because it's handled by the overloads above.

Reply:
removed in [e112344](https://github.com/PrefectHQ/prefect/pull/14197/commits/e112344a950c00113253b91778939d9165f23978)